File: Weapon/GunshotWeapon/GunshotWeapon.py
import sys
sys.path.append('../')
from Weapon import *


class GunshotWeapon(Weapon):
    __caliber = None
    __range = None

    @property
    def caliber(self):
        return self.__caliber

    @caliber.setter
    def caliber(self, value):
        self.__caliber = value

    @property
    def range(self):
        return self.__range

    @range.setter
    def range(self, value):
        self.__range = value

File: Weapon/GunshotWeapon/GunshotWeapon.php
<?php
namespace Weapon;



abstract class GunshotWeapon extends Weapon{
    private $caliber;
    private $range;

    final public function setRange(int $range)
    {
        $this->range = $range;
    }

    final public function setCaliber(float $caliber){
        $this->caliber = $caliber;
    }
    final public function getCaliber(){
        return $this->caliber;
    }
    final public function getRange(){
        return $this->range;
    }
}